go/consensus/tendermint/apps/registry/state/state.go

```diff
@@ -443,6 +375,46 @@ func (s *ImmutableState) NodeStatus(ctx context.Context, id signature.PublicKey)
 	return &status, nil
 }
 
+// GetEntityNodes returns nodes registered by given entity.
+// Note that this returns both active and expired nodes.
+func (s *ImmutableState) GetEntityNodes(ctx context.Context, id signature.PublicKey) ([]*node.Node, error) {
+	it := s.is.NewIterator(ctx)
+	defer it.Close()
+
+	hID := keyformat.PreHashed(id.Hash())
+
+	var nodes []*node.Node
+	for it.Seek(signedNodeByEntityKeyFmt.Encode(&id)); it.Valid(); it.Next() {
+		var hEntityID keyformat.PreHashed
+		var hNodeID keyformat.PreHashed
+
+		if !signedNodeByEntityKeyFmt.Decode(it.Key(), &hEntityID, &hNodeID) || !hEntityID.Equal(&hID) {
+			break
+		}
+
+		rawSignedNode, err := s.is.Get(ctx, signedNodeKeyFmt.Encode(&hNodeID))
+		if err != nil {
+			return nil, abciAPI.UnavailableStateError(err)
+		}
+		if rawSignedNode == nil {
+			return nil, registry.ErrNoSuchNode
```

Comment:
Doesn't this indicate state corruption as if the node is removed the `signedNodeByEntityKey` index should also be updated? In this case wrap it in `abciAPI.UnavailableStateError(...)`.

Reply:
Good catch, I'll add that.